src/PSRule.Rules.Azure/rules/Azure.MariaDB.Rule.ps1

```diff
@@ -126,6 +126,14 @@ Rule 'Azure.MariaDB.FirewallRuleCount'-Ref 'AZR-000343' -Type 'Microsoft.DBforMa
     Reason($LocalizedData.DBServerFirewallRuleCount, $firewallRules.Length, 10).PathPrefix('resources')
 }
 
+# Synopsis: Determine if there is an excessive number of permitted IP addresses.
+Rule 'Azure.MariaDB.FirewallIPRange' -Ref 'AZR-000344' -Type 'Microsoft.DBforMariaDB/servers' -Tag @{ release = 'GA'; ruleSet = '2022_12'; } {
+    $summary = GetIPAddressSummary
+
+    $Assert.LessOrEqual($summary, 'Public', 10).
```

Comment:
@BenjaminEngeset There seems to be a bug with the `LessOrEqual` method not handling uint64 types correctly. Try this.

```suggestion
    [int]$public = [int]$summary.Public
    $Assert.LessOrEqual($public, '.', 10).
```